src/TLuaInterpreter.cpp

```diff
@@ -11728,35 +11728,264 @@ int TLuaInterpreter::sendSocket( lua_State * L )
-    return 0;
+    QString target = who.c_str();
+    QString msg = text.c_str();
+    Host* pHost = &getHostFromLua(L);
+    if ( !mudlet::self()->mpIrcClientMap[pHost] ) {
+        // create a new irc client if one isn't ready.
+        mudlet::self()->mpIrcClientMap[pHost] = new dlgIRC(pHost);
+        mudlet::self()->mpIrcClientMap[pHost]->raise();
+        mudlet::self()->mpIrcClientMap[pHost]->show();
+    }
+
+    // wait for our client to be ready before sending messages.
+    if (!mudlet::self()->mpIrcClientMap[pHost]->mReadyForSending) {
+        lua_pushboolean(L, false);
+        return 2;
+    }
+
+    lua_pushboolean(L, mudlet::self()->mpIrcClientMap[pHost]->sendMsg(target, msg));
+    return 1;
+}
+
+/** ircGetNick();
+ *  Returns a string containing the IRC client nickname.
+ */
+int TLuaInterpreter::ircGetNick(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QString nick;
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        nick = mudlet::self()->mpIrcClientMap[pHost]->getNickName();
+    } else {
+        nick = dlgIRC::readIrcNickName(pHost);
+    }
+
+    lua_pushstring(L, nick.toUtf8().constData());
+    return 1;
+}
+
+/** ircGetServer();
+ *  Returns a pair containing the IRC server and port.
+ */
+int TLuaInterpreter::ircGetServer(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QString hname;
+    int hport;
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        hname = mudlet::self()->mpIrcClientMap[pHost]->getHostName();
+        hport = mudlet::self()->mpIrcClientMap[pHost]->getHostPort();
+    } else {
+        hname = dlgIRC::readIrcHostName(pHost);
+        hport = dlgIRC::readIrcHostPort(pHost);
+    }
+
+    lua_pushstring(L, hname.toUtf8().constData());
+    lua_pushinteger(L, hport);
+    return 2;
+}
+
+/** ircGetChannels();
+ *  Returns a string containing channels separated by a space.
+ *  If a client is active the list contains channels currently joined.
+ *  Otherwise the list is read from IRC client settings.
+ */
+int TLuaInterpreter::ircGetChannels(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QStringList channels;
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        channels = mudlet::self()->mpIrcClientMap[pHost]->getChannels();
+    } else {
+        channels = dlgIRC::readIrcChannels(pHost);
+    }
+
+    lua_pushstring(L, channels.join(" ").toUtf8().constData());
+    return 1;
+}
+
+/** ircGetConnectedHost()
+ *  Returns the Hostname of the connected IRC Server, as provided by the server itself.
+ *  This will return an empty string until the IRC Client has connected and received
+ *  a hostname message from the server.
+ */
+int TLuaInterpreter::ircGetConnectedHost(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QString cHostName = "";
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        cHostName = mudlet::self()->mpIrcClientMap[pHost]->getConnectedHost();
+    }
+
+    lua_pushstring(L, cHostName.toUtf8().constData());
```

Comment:
The lua way would be to return false+nil, because then you can do:

```lua
local host = ircGetConnectedHost()
if not host then return end
```

Whereas if you get a blank string you have to do:

```lua
local host = ircGetConnectedHost()
if host == "" then return end
```

Not as tidy hey?